### PR TITLE
Simplify README structure and language

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

None, as it's a documentation fix

> Is there anything in the PR that needs further explanation?

A lot of stylelint's users are not native English speakers. This PR tries to simplify the structure and the the language of the README (and one or two other pages). The British Government has good [research and guidance](https://www.gov.uk/guidance/content-design/writing-for-gov-uk) on writing for the web, and I've tried to incorporate it into this PR.

I had a cursory look through the other pages, but I decided to focus on main README for now as it has the high information density.

### CODE REVIEW

File: docs/user-guide/cli.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 stylelint is an [npm package](https://www.npmjs.com/package/stylelint). Install it using:
 
 ```shell
-npm install -g stylelint
+npm install stylelint --save-dev save-prefix='~'
```
Comment (hudochenkov) [inline docs/user-guide/cli.md]: What is `save-prefix='~'`?

Comment (jeddy3) [reply docs/user-guide/cli.md]: Oops, I thought [this config option](https://docs.npmjs.com/misc/config#save-prefix) was available as a flag. But it doesn't seem to be. I'll change it back.

**Comment by jeddy3** [comment]
@hudochenkov Done.

**Comment by jeddy3** [comment]
If we can get this in today, then I should be able to do a release tonight.

**Comment by ntwb** [comment]
Hey @stylelint/contributors can w get another approval on this PR please 🙇

File: README.md [hunk 2]
```diff
@@ -1,124 +1,140 @@
...
 
--   [User guide](docs/user-guide.md) - Usage, configuration, FAQ and complementary tools.
--   [Developer guide](docs/developer-guide.md) - Contributing to stylelint and writing your own plugins & formatters.
+-   [user guide](docs/user-guide.md) - how to use and configure stylelint
+-   [developer guide](docs/developer-guide.md) - how to develop for stylelint
 
 ## Need help?
 
-If you're looking for help or have a support question, then check out our [FAQ](docs/user-guide/faq.md) first. If the answer to your problem isn't there, then go to [stackoverflow](https://stackoverflow.com/questions/tagged/stylelint). stackoverflow is a huge Question and Answer community, and tagging your post there with "stylelint" will catch the stylelint team's attention.
+Read our [FAQ](docs/user-guide/faq.md) first. If the answer to your problem isn't there, then post it to [stackoverflow](https://stackoverflow.com/questions/tagged/stylelint).
```
Comment (montezume) [inline README.md]: nitpick: post it `on` stackoverflow sounds better to me.

File: docs/user-guide/about-rules.md [hunk 3]
```diff
@@ -2,24 +2,30 @@
...
 
 ## About rule names
 
--   Made of lowercase words separated by hyphens.
--   Split into two parts:
-    -   The first describes what [*thing*](http://apps.workflower.fi/vocabs/css/en) the rule applies to.
-    -   The second describes what the rule is checking.
+Rule names are:
+
+-   made of lowercase words separated by hyphens
```
Comment (montezume) [inline docs/user-guide/about-rules.md]: made up of?

**Pull request merged**